batch7crud-roll10/src/main/java/com/lftechnology/batch7crud/controller/StudentController.java

```diff
@@ -0,0 +1,209 @@
+      }
+
+      list(req, resp, page);
+
+    } else {
+
+      if ("create".equalsIgnoreCase(params[1])) {
+        create(req, resp);
+
+      } else if ("edit".equalsIgnoreCase(params[2])) {
+        Integer studentId = TypeCaster.toInt(params[1]);
+        if (studentId == null) {
+          showPageNotFound(req, resp);
+        } else {
+          edit(req, resp, studentId);
+        }
+
+      }
+    }
+
+  }
+
+  @Override
+  protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+    String[] params = params(req);
+
+    if ("create".equalsIgnoreCase(params[1])) {
+      createProcess(req, resp);
+
+    } else if ("edit".equalsIgnoreCase(params[2])) {
+      Integer studentId = TypeCaster.toInt(params[1]);
+      if (studentId == null) {
+        showPageNotFound(req, resp);
+      } else {
+        editProcess(req, resp, studentId);
+      }
+
+    } else if ("delete".equalsIgnoreCase(params[2])) {
+      Integer studentId = TypeCaster.toInt(params[1]);
+      if (studentId == null) {
+        showPageNotFound(req, resp);
+      } else {
+        deleteProcess(req, resp, studentId);
+      }
+    }
+  }
+
+  private void list(HttpServletRequest req, HttpServletResponse resp, int page) throws ServletException, IOException {
+    try {
+      Integer offset = (page - 1) * TOTAL_DATA_TO_FETCH;
+      List<Student> studentList = studentService.fetch(offset, TOTAL_DATA_TO_FETCH);
+      Integer totalRecord = studentService.fetchTotalRecordNumber();
+      Double totalPages = ceil(totalRecord * 1.0 / TOTAL_DATA_TO_FETCH);
+
+      req.setAttribute("studentList", studentList);
+      req.setAttribute("totalPages", totalPages.intValue());
+      req.setAttribute("currentPage", page);
+      req.getServletContext().getRequestDispatcher("/WEB-INF/views/students.jsp").forward(req, resp);
+    } catch (DataException e) {
+      LOGGER.log(Level.SEVERE, e.getMessage(), e);
+      showServerError(req, resp, e);
+    }
+  }
+
+  private void create(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+    req.getServletContext().getRequestDispatcher("/WEB-INF/views/createStudent.jsp").forward(req, resp);
+  }
+
+  private void createProcess(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+    String name = req.getParameter("name");
+    String address = req.getParameter("address");
+    String dob = req.getParameter("dob");
+    String department = req.getParameter("department");
+    String batch = req.getParameter("batch");
+    String rollText = req.getParameter("roll");
+
+    try {
+      DateFormat dateFormat = new SimpleDateFormat("yyyy-mm-dd", Locale.ENGLISH);
+      Date date = dateFormat.parse(dob);
+      Integer roll = Integer.parseInt(rollText);
+
+      Student student = new Student();
+      student.setBatch(batch);
+      student.setDepartment(department);
+      student.setRoll(roll);
+
```

Comment:
Instead create another method private Student constructStudentFromRequest(), method is too long.